dockerfiles/init/modules/openshift/files/scripts/ocp.sh

```diff
@@ -0,0 +1,291 @@
+    if [ ! -z "$ocp_registry_container_id" ];then
+        ocp_registry_container_status=$(docker inspect "$ocp_registry_container_id" | $JQ_BINARY .[0] | $JQ_BINARY -r '.State.Status')
+    else
+        return 1
+    fi
+    if [[ "${ocp_registry_container_status}" == "running" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+wait_ocp() {
+  OCP_BOOT_TIMEOUT=120
+  echo "[OCP] wait for ocp full boot..."
+  ELAPSED=0
+  until ocp_is_booted; do
+    if [ ${ELAPSED} -eq "${OCP_BOOT_TIMEOUT}" ];then
+        echo "OCP didn't started in $OCP_BOOT_TIMEOUT secs, exit"
+        exit 1
+    fi
+    sleep 2
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+run_ocp() {
+    $OC_BINARY cluster up --public-hostname="${OC_PUBLIC_HOSTNAME}" --routing-suffix="${OC_PUBLIC_IP}.nip.io"
+    wait_ocp
+}
+
+deploy_che_to_ocp() {
+    #Repull init image only if DEFAULT_IMAGE_PULL_POLICY is set to Always
+    if [ $DEFAULT_IMAGE_PULL_POLICY == "Always" ]; then
+        docker pull "$IMAGE_INIT"
+    fi
+    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:nightly destroy --quiet --skip:pull --skip:nightly
+    docker run -t --rm -v /var/run/docker.sock:/var/run/docker.sock -v "${CONFIG_DIR}":/data -e IMAGE_INIT="$IMAGE_INIT" -e CHE_MULTIUSER="$CHE_MULTIUSER" eclipse/che-cli:nightly config --skip:pull --skip:nightly
+    cd "${CONFIG_DIR}/instance/config/openshift/scripts/"
+    bash deploy_che.sh
+    wait_until_server_is_booted
+}
+
+server_is_booted() {
+  PING_URL="http://che-$OPENSHIFT_NAMESPACE_URL"
+  HTTP_STATUS_CODE=$(curl -I -k "${PING_URL}" -s -o /dev/null --write-out '%{http_code}')
+  if [[ "${HTTP_STATUS_CODE}" = "200" ]] || [[ "${HTTP_STATUS_CODE}" = "302" ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+wait_until_server_is_booted() {
+  SERVER_BOOT_TIMEOUT=300
+  echo "[CHE] wait CHE pod booting..."
+  ELAPSED=0
+  until server_is_booted || [ ${ELAPSED} -eq "${SERVER_BOOT_TIMEOUT}" ]; do
+    sleep 2
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+check_workspace_status() {
+  STATUS_URL="http://che-${OPENSHIFT_NAMESPACE_URL}/api/workspace/${ws_id}"
+  WS_STATUS=$(curl -s "${STATUS_URL}" | $JQ_BINARY -r '.status')
+  if [[ "${WS_STATUS}" == *"$1"* ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+wait_workspace_status() {
+  STATUS=$1
+  WS_BOOT_TIMEOUT=300
+  echo "[TEST] wait che workspace status is ${STATUS}..."
+  ELAPSED=0
+  until check_workspace_status "${STATUS}" || [ ${ELAPSED} -eq "${WS_BOOT_TIMEOUT}" ]; do
+    sleep 2
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+run_test() {
+    #TODO FIX for multi user need to handle auth
+    echo "[TEST] run CHE workspace test"
+    ws_name="ocp-test-$(date +%s)"
+
+    # create workspace
+    ws_create=$(curl -s "http://che-${OPENSHIFT_NAMESPACE_URL}/api/workspace?namespace=che&attribute=stackId:java-centos" \
```

Comment:
Can you move this into a separate file and load content from it?

Reply:
for now I think I will remove run_test because it is outdated abit and was created for CHE5 not for CHE6

Reply:
removed from script